fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/highlight/LuceneScaleTest.java

```diff
@@ -251,50 +257,77 @@ void runPerfTest() throws IOException, ExecutionException, InterruptedException
+                        mergeIndexes(indexesRequireMerge, mergeTimer, mergeCsv, testStartMillis, dataModel);
                     }
-                    updateCsv("Did insert", dataModel, insertsCsv, startMillis, testStartMillis, Map.of());
+                    updateCsv("Did insert", dataModel, insertsCsv, startMillis, testStartMillis, Map.of(), timer);
                 }
                 if (Config.COMMANDS_TO_RUN.contains(Command.Update)) {
                     timer.reset();
                     startMillis = System.currentTimeMillis();
                     for (int j = 0; j < operationCount; j++) {
-                        dataModel.updateRecords(updatesPerContext);
+                        final Set<Index> indexesRequireMerge = dataModel.updateRecords(updatesPerContext);
+                        mergeIndexes(indexesRequireMerge, mergeTimer, mergeCsv, testStartMillis, dataModel);
                     }
                     updateCsv("Did updates", dataModel, updatesCsv, startMillis, testStartMillis,
                             Map.of("updatesPerContext", updatesPerContext,
-                                    "updateBatches", operationCount));
+                                    "updateBatches", operationCount), timer);
                 }
                 if (Config.COMMANDS_TO_RUN.contains(Command.Search)) {
                     timer.reset();
                     startMillis = System.currentTimeMillis();
                     for (int j = 0; j < operationCount; j++) {
                         dataModel.search();
                     }
-                    updateCsv("Did Search", dataModel, searchesCsv, startMillis, testStartMillis, Map.of());
+                    updateCsv("Did Search", dataModel, searchesCsv, startMillis, testStartMillis, Map.of(), timer);
                 }
                 dataModel.updateSearchWords();
 
             }
         }
     }
 
-    private void updateCsv(final String logTtl, final DataModel dataModel, final PrintStream csvPrintStream,
-                           final long startMillis, final long testStartMillis, final Map<?, ?> additionalKeysAndValues) {
-        final Map<String, Number> keysAndValues = timer.getKeysAndValues();
+    private void mergeIndexes(final Set<Index> indexesRequireMerge, final FDBStoreTimer mergeTimer,
+                              @Nullable final PrintStream mergeCsv, final long testStartMillis, final DataModel dataModel) {
+        long startMillis = System.currentTimeMillis();
+        for (final Index index : indexesRequireMerge) {
+            final OnlineIndexer onlineIndexer = OnlineIndexer.newBuilder()
+                    .addTargetIndex(index)
+                    .setRecordStore(recordStore)
+                    .setTimer(mergeTimer)
+                    .build();
+            onlineIndexer.mergeIndex();
+        }
+
+        if (mergeCsv != null) {
+            updateCsv("Did insert", dataModel, mergeCsv, startMillis, testStartMillis, Map.of(
```

Comment:
Whoops, this is wrong, and we are preserving the timer across multiple merges, skewing metrics. Will push an update in a sec

Reply:
ok, I pushed the change, and it appeared in GitHub, but the PR doesn't appear to be updating... https://github.com/ScottDugas/fdb-record-layer/blob/ca323d6d3d1fca2d9e599d7a067de71f0c6ab938/fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/highlight/LuceneScaleTest.java#L308